lisa/features/gpu.py

```diff
@@ -126,23 +142,82 @@ def check_support(self) -> None:
         if not self._is_supported():
             raise SkippedException(f"GPU is not supported with distro {self._node.os}")
 
-    def install_compute_sdk(
-        self, driver: ComputeSDK = ComputeSDK.CUDA, version: str = ""
-    ) -> None:
+    def install_compute_sdk(self, version: str = "") -> None:
         # install GPU dependencies before installing driver
         self._install_gpu_dep()
 
+        if isinstance(self._node.os, Redhat):
+            # install LIS driver if not already installed.
+            self._node.tools[LisDriver]
+
         # install the driver
-        if driver == ComputeSDK.GRID:
-            if version == "":
-                version = DEFAULT_GRID_DRIVER_URL
-            self._install_grid_driver(version)
-        elif driver == ComputeSDK.CUDA:
-            if version == "":
-                version = "10.1.105-1"
-            self._install_cuda_driver(version)
-        else:
+        supported_driver = self._get_supported_driver()
+        for driver in supported_driver:
+            if driver == self.ComputeSDK.GRID:
+                if version == "":
+                    version = DEFAULT_GRID_DRIVER_URL
+                    self._install_grid_driver(version)
+                    self.gpu_vendor.add("nvidia")
+            elif driver == self.ComputeSDK.CUDA:
+                if version == "":
+                    version = "10.1.105-1"
+                    self._install_cuda_driver(version)
+                    self.gpu_vendor.add("nvidia")
+
+    def validate_gpu_count_with_lsvmbus(self, expected_count: int) -> None:
+        # Device Id for Tesla GPU SKUs
+        tesla_device_id_pattern = "47505500"
+        # Device Id for A100-SXM4 GPU SKUs
+        a100_device_id_pattern = "444500000"
+
+        lsvmbus_device_count = 0
+        gpu_bridge_count = 0
+
+        lsvmbus_tool = self._node.tools[Lsvmbus]
+        device_list = lsvmbus_tool.get_device_channels_from_lsvmbus()
+        for device in device_list:
+            if tesla_device_id_pattern in device.device_id:
+                lsvmbus_device_count += 1
+            elif a100_device_id_pattern in device.device_id:
+                lsvmbus_device_count += 1
+                gpu_bridge_count = 6
```

Comment:
The number can be put with gpu names together. It doesn't need to be named like gpu_bridge_count, it can be "extra_device_count". It should be more general.